tests/test_utils.py

```diff
@@ -765,3 +765,166 @@ def test_csv_latin_1_encoding(self):
+        with open(csv_file, 'w', encoding='utf-8-sig') as file_obj:
+            file_obj.write(textwrap.dedent(
+                """
+                header1,HEADER2,header3
+                1,2,bar
+                4,5,FOO
+                """
+            ).strip())
+        # confirm that the file has the BOM prefix
+        with open(csv_file, 'rb') as file_obj:
+            self.assertTrue(file_obj.read().startswith(codecs.BOM_UTF8))
+
+        df = utils.read_csv_to_dataframe(csv_file)
+        # assert the BOM prefix was correctly parsed and skipped
+        self.assertEqual(df.columns[0], 'header1')
+        self.assertEqual(df['HEADER2'][1], 5)
+
+    def test_non_utf8_encoding(self):
+        """Test that non-UTF8 encoding doesn't raise an error"""
+        from natcap.invest import utils
+
+        csv_file = os.path.join(self.workspace_dir, 'csv.csv')
+
+        # encode with ISO Cyrillic, include a non-ASCII character
+        with open(csv_file, 'w', encoding='iso8859_5') as file_obj:
+            file_obj.write(textwrap.dedent(
+                """
+                header,
+                fЮЮ,
+                bar
+                """
+            ).strip())
+        df = utils.read_csv_to_dataframe(csv_file)
+        # the default engine='python' should replace the unknown characters
+        # different encodings of replacement character depending on the system
+        self.assertTrue(df['header'][0] in ['f\xce\xce', 
+            'f\N{REPLACEMENT CHARACTER}\N{REPLACEMENT CHARACTER}'])
+        self.assertEqual(df['header'][1], 'bar')
+
+    def test_override_default_encoding(self):
+        """Test that you can override the default encoding kwarg"""
+        from natcap.invest import utils
+
+        csv_file = os.path.join(self.workspace_dir, 'csv.csv')
+
+        # encode with ISO Cyrillic, include a non-ASCII character
+        with open(csv_file, 'w', encoding='iso8859_5') as file_obj:
+            file_obj.write(textwrap.dedent(
+                """
+                header,
+                fЮЮ,
+                bar
+                """
+            ).strip())
+        df = utils.read_csv_to_dataframe(csv_file, encoding='iso8859_5')
+        # with the encoding specified, special characters should work
+        self.assertEqual(df['header'][0], 'fЮЮ')
+        self.assertEqual(df['header'][1], 'bar')
+
+    def test_other_kwarg(self):
+        """Any other kwarg should be passed to pandas.read_csv"""
+        from natcap.invest import utils
+
+        csv_file = os.path.join(self.workspace_dir, 'csv.csv')
+
+        with open(csv_file, 'w') as file_obj:
+            file_obj.write(textwrap.dedent(
+                """
+                h1;h2;h3
+                a;b;c
+                d;e;f
+                """
+            ).strip())
+        # using sep=None with the default engine='python',
+        # it should infer what the separator is
+        df = utils.read_csv_to_dataframe(csv_file, sep=None)

```

Comment:
Ooo thanks for reminding me of this!  I'm sorry I had forgotten about it, but I would like to make a case for assuming `sep=None`, at least as a default.  Many users of InVEST (especially when the computer's language is not English) will export Excel tables to CSV but then their separator will be semicolons rather than commas.  Passing `sep=None` fixes that, and this is exactly what `build_lookup_from_csv`does.  What do you think?

Reply:
That sounds like a good default to have!